frame/pablo/src/lib.rs

```diff
@@ -322,6 +322,43 @@ pub mod pallet {
 		type TWAPInterval: Get<MomentOf<Self>>;
 
 		type WeightInfo: WeightInfo;
+
+		fn redeemable_assets_for_given_lp_tokens(
+			pool: PoolConfigurationOf<Self>,
+			pool_account: Self::AccountId,
+			lp_amount: Self::Balance,
+		) -> Result<(Self::Balance, Self::Balance), DispatchError> {
```

Comment:
What's preventing us from using the same `RedeemableAssets` data structure here as well?

Reply:
We need a test for this function

Reply:
> What's preventing us from using the same `RedeemableAssets` data structure here as well?

What is need for that? Given RedeemableAssets will have map now. Why to iterate over map?

Reply:
Not sure I understand, which place exactly you are referring to where iteration occurs?

Reply:
I am using this new function in remove_liquidity(), there I need to query map to get base/quote asset balance.

Reply:
Since other pallets integrate with Pablo using this trait, this would save us from some refactorings in the future across multiple pallets when there are more than two redeemable assets in the pool. Also having the assetId instead of just balances to avoid mistakes.

Reply:
Hmm I also notice that this particular function follows a different pattern than what we follow for other trait impls where the `Amm` trait impl is used by the `Config` trait instead of the other way around. I think we may change that around if there is no blocker for that :)

Reply:
yep. i fixed all that in lending to be like that.